plan/validator.go

```diff
@@ -444,12 +444,27 @@ func checkColumn(colDef *ast.ColumnDef) error {
 				return types.ErrIllegalValueForType.GenByArgs(types.TypeStr(tp.Tp), str)
 			}
 		}
+	case mysql.TypeDate:
+		for _, columnOpt := range colDef.Options {
+			if columnOpt.Tp == ast.ColumnOptionDefaultValue && isNowBuiltinFunc(columnOpt.Expr) {
+				return types.ErrInvalidDefault.GenByArgs(colDef.Name.Name.O)
+			}
+		}
 	default:
 		// TODO: Add more types.
 	}
 	return nil
 }
 
+func isNowBuiltinFunc(expr ast.ExprNode) bool {
```

Comment:
This function checks `now` symbolic, it doesn't have to be exactly `now`.
Use `isNowSymFunc` is better, like the one defined in parser.